src/vsr/client.zig

```diff
@@ -188,8 +189,21 @@ pub fn ClientType(
             self.message_bus.deinit(allocator);
         }
 
-        pub fn on_message(message_bus: *MessageBus, message: *Message) void {
+        pub fn on_messages(message_bus: *MessageBus, buffer: *ReceiveBuffer) void {
             const self: *Client = @fieldParentPtr("message_bus", message_bus);
+            while (buffer.consume_message(self.message_bus.pool)) |message| {
+                defer self.message_bus.unref(message);
+
+                if (message.header.cluster != self.cluster) {
+                    buffer.invalidate(.header_cluster);
+                    return;
+                }
+                self.on_message(message);
+                if (self.evicted) break;
```

Comment:
I _think_ it was actually possible to hit the `assert(!self.evicted)` assert in `on_message` before --- production message bus _can_  push several messages onto a client in one go.